grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/redisqueue/RedisTaskStorage.java

```diff
@@ -102,17 +117,24 @@ public TaskState getState(TaskId id) throws GraknBackendException {
     @Override
     public boolean containsTask(TaskId id) {
         try(Jedis jedis = redis.getResource()){
-            String value = jedis.get(id.getValue());
+            String value = jedis.get(encodeKey.apply(id.getValue()));
             return value != null;
         }
     }
 
     @Override
-    public Set<TaskState> getTasks(TaskStatus taskStatus, String taskClassName, String createdBy,
-            EngineID runningOnEngine, int limit, int offset) {
+    public Set<TaskState> getTasks(@Nullable TaskStatus taskStatus, @Nullable String taskClassName,
+            @Nullable String createdBy, @Nullable EngineID runningOnEngine, int limit, int offset) {
         try (Jedis jedis = redis.getResource()) {
-            // TODO change structure of task keys with a prefix. Can we do better than filtering?
-            Stream<TaskState> stream = jedis.keys("*-*-*-*").stream().map(value -> (TaskState) deserialize(Base64.getDecoder().decode(value)));
+            Stream<TaskState> stream = jedis.keys(PREFIX + "*").stream().map(key -> {
```

Comment:
Should be fine™ 

"While the time complexity for this operation is O(N), the constant times are fairly low. For example, Redis running on an entry level laptop can scan a 1 million key database in 40 milliseconds."

Considering it's just to scan the tasks, we should rarely get to 1 million tasks in the queue